### ISSUE TITLE
How does someone view the differences between two "releases" of WCAG 2.2 guidelines?

### ISSUE THREAD

**Issue by OwenEdwards** [w3c/wcag#1808]
Looking at the last two "releases" of the WCAG 2.2 Guidelines:

https://www.w3.org/TR/2020/WD-WCAG22-20200811/

https://www.w3.org/TR/2021/WD-WCAG22-20210513/

The May 2021 release has a link to [Commit history](https://github.com/w3c/wcag/commits/gh-pages), which does lead back to this GitHub repo and shows all the commits with dates and the usual brief descriptions. But, since the "20200811" and "20210513" are not tagged as releases in the GitHub repo, the features of GitHub cannot be used to easily see "what has changed between the 20200811 and 20210513 release". 

There's a section in the document called "[5.1 New Features in WCAG 2.2](https://www.w3.org/TR/2021/WD-WCAG22-20210513/#new-features-in-wcag-2-2)", but this describes changes since WCAG 2.1. Is there a part of the document that describes changes since the last *release* of WCAG 2.2 on 20200811, or a way to use GitHub to see the differences?

If not, I suggest using the "[Releases](https://github.com/w3c/wcag/releases)" feature to allow people to easily compare releases, (like in video.js for example - see https://github.com/videojs/video.js/releases, and e.g. https://github.com/videojs/video.js/compare/v7.11.8...v7.12.1).

**Comment by alastc** [comment]
There is a 'diff' service, which generates this:
https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2FTR%2F2020%2FWD-WCAG22-20200811%2F&doc2=https%3A%2F%2Fwww.w3.org%2FTR%2F2021%2FWD-WCAG22-20210513%2F

However, I'd give it a couple of days as there was a bug in the publication script which removed all the understanding document links (face-palm gif).

There will be a new publication tomorrow or Friday, and I can update the diff link then. (Or just use https://services.w3.org/htmldiff yourself.) 

I think the problem with a github approach is that you'd get a lot of other documents which have changed as well, as the repo includes all the technique/understanding documents for all 2.x versions.

**Comment by OwenEdwards** [comment]
@alastc thanks for the info about the "diff" service, that is helpful.

But I do wonder how the "Commit history" link under "Participate" in the Guidelines document is meant to be helpful - it takes you to a GitHub page where every commit is by the "traviscibot" with a  description of "Deploy w3c/wcag to github.com/w3c/wcag.git:gh-pages". If the "gh-pages" branch of the repo is where the "Commit history" can be viewed, wouldn't having release tagging on that branch allow people to see an overview of changes, rather than having to dig into each commit which doesn't have a descriptive description?

I'm not clear exactly what the "traviscibot" is doing with that branch, but it seems like it may be focused on changes just to the "guidelines/22/index.html" file, which is exactly what I'm saying would be useful to see a diff of.

**Comment by kfranqueiro** [comment]
The commit history link in the latest publications of WCAG 2.1 and WCAG 2.2 now link to the full history of the main branch, which unfortunately still isn't ideal, but if I recall correctly, is a limitation of ReSpec configuration.

The most relevant github commit history links are as follows:

[`guidelines` folder for WCAG 2.2](https://github.com/w3c/wcag/commits/main/guidelines) (on the `main` branch)
[`guidelines` folder for WCAG 2.1](https://github.com/w3c/wcag/commits/WCAG-2.1/guidelines) (on the `WCAG-2.1` branch, which is specifically used only for the TR document)

Judging by [the documentation of ReSpec's `github` property](https://github.com/speced/respec/wiki/github), it should be possible to point 2.1 to the correct branch. Pointing to a subfolder does not seem to be supported.

(Note: ReSpec's documentation also seems to reflect `gh-pages` being the default branch, but [the source code](https://github.com/speced/respec/blob/19b2c0e2src/core/github.js#L97) suggests otherwise, which may be how that particular behavior originally mentioned in this issue became resolved.)

**Comment by kfranqueiro** [comment]
To wrap up the questions in this issue:

## Links to commit history in details

My [previous comment](https://github.com/w3c/wcag/issues/1808#issuecomment-2695133515) refers to behavior that was likely introduced to respec some time after this issue was first raised, so there is now a link to commit history within the details at the top of newer publications of WCAG 2.1 and 2.2.

The commit history links point to the appropriate branch in each of the most recent publications for 2.1 and 2.2. Those links will show the full repo history (including changes to informative docs); see the links in my previous comment for URLs to the `guidelines` folder for each.

## Tags

The 2 most recent publications of each of WCAG 2.1 and WCAG 2.2 have corresponding [tags](https://github.com/w3c/wcag/tags) in this repository. This provides a couple of options, but as far as I'm aware, each approach has a trade-off:

- You can view the commit history of only the `guidelines` folder leading up to any of those tags, e.g. https://github.com/w3c/wcag/commits/WCAG22-20241212/guidelines
  - Trade-off: there's no way to set a range or start point, so you'd have to keep the previous tag's commit hash in mind and stop when it appears in the list
- You can view the diff between the tags, e.g. https://github.com/w3c/wcag/compare/WCAG22-20231005...WCAG22-20241212
  - Trade-off: there's no way to limit this to only the `guidelines` subfolder, which is primarily an issue for 2.2; the `WCAG-2.1` branch typically only receives `guidelines`-related backports now

## HTML Diff

As Alastair mentioned, you can enter the URLs of the older and newer version into https://services.w3.org/htmldiff to generate an HTML diff view.
